tests/test_json_schema.py

```diff
@@ -1382,8 +1383,12 @@ class Model(BaseModel):
     class MyGenerator(GenerateJsonSchema):
         ignored_warning_kinds = ()
 
-    with pytest.warns(PydanticJsonSchemaWarning, match=warning_match):
-        model_schema = Model.model_json_schema(schema_generator=MyGenerator)
+    with warnings.catch_warnings():
+        # we need to explicitly ignore the other warning in pytest-8
+        # TODO: rewrite it to use two nested pytest.warns() when pytest-7 is no longer supported
+        warnings.simplefilter("ignore")
```

Comment:
Same question here, what warnings are being raised?

Reply:
It's in the commit message. When we're matching "Default value .* is not JSON serializable", it's emitting "Cannot generate a JsonSchema", and the other way around. I wasn't able to find a way to successfully match both simultaneously in pytest-7.

Reply:
@mgorny,

What about something like this:

```py
import warnings
import pytest

def test_multiple_warnings():
    with pytest.warns((UserWarning, RuntimeWarning)) as record:
        warnings.warn("This is a user warning", UserWarning)
        warnings.warn("This is a runtime warning", RuntimeWarning)

    assert len(record) == 2  # check that two warnings were raised
    assert isinstance(record[0].message, UserWarning)  # check the type of the first warning
    assert str(record[0].message) == "This is a user warning"  # check the first warning message
    assert isinstance(record[1].message, RuntimeWarning)  # check the type of the second warning
    assert str(record[1].message) == "This is a runtime warning"  # check the second warning message
```

Obviously this would be specialized to the case for the warnings raised above